ql/src/java/org/apache/hadoop/hive/ql/parse/ColumnStatsSemanticAnalyzer.java

```diff
@@ -304,15 +287,199 @@ public static String genRewrittenQuery(Table tbl, List<String> colNames, HiveCon
     }
 
     String rewrittenQuery = rewrittenQueryBuilder.toString();
-    rewrittenQuery = new VariableSubstitution(new HiveVariableSource() {
-      @Override
-      public Map<String, String> getHiveVariable() {
-        return SessionState.get().getHiveVariables();
-      }
-    }).substitute(conf, rewrittenQuery);
+    rewrittenQuery = new VariableSubstitution(
+        () -> SessionState.get().getHiveVariables()).substitute(conf, rewrittenQuery);
     return rewrittenQuery;
   }
 
+  private static void genComputeStats(StringBuilder rewrittenQueryBuilder, HiveConf conf,
+      int pos, String columnName, TypeInfo typeInfo) throws SemanticException {
+    Preconditions.checkArgument(typeInfo.getCategory() == Category.PRIMITIVE);
+    ColumnStatsType columnStatsType =
+        ColumnStatsType.getColumnStatsType((PrimitiveTypeInfo) typeInfo);
+    // The first column is always the type
+    // The rest of columns will depend on the type itself
+    int size = columnStatsType.getColumnStats().size() - 1;
+    for (int i = 0; i < size; i++) {
+      ColumnStatsField columnStatsField = columnStatsType.getColumnStats().get(i);
+      appendStatsField(rewrittenQueryBuilder, conf, columnStatsField, columnStatsType,
+          columnName, pos);
+      rewrittenQueryBuilder.append(", ");
```

Comment:
instead of duplicating the loop body; a conditional for the "append" might be better; it's easier to write than describe; I'm thinking something like:
```
for(a : x) {
if(!first) append(",");
body();
}
```

Reply:
Done.